src/AbstractDictionary.jl

```diff
@@ -363,23 +363,32 @@ end
 empty_type(::Type{<:AbstractDictionary}, ::Type{I}, ::Type{T}) where {I, T} = Dictionary{I, T}
 Base.empty(dict::AbstractDictionary, ::Type{I}, ::Type{T}) where {I, T} = empty_type(typeof(dict), I, T)()
 
-function Base.merge(d1::AbstractDictionary{K1, T1}, d2::AbstractDictionary{K2, T2}) where {K1, T1, K2, T2}
+function Base.merge(d1::AbstractDictionary, others::AbstractDictionary...)
     # Note: need to copy the keys
-    out = similar(copy(keys(d1), promote_type(K1, K2)), promote_type(T1, T2))
+    K = promote_type(keytype(d1), keytype.(others)...)
+    T = promote_type(valtype(d1), valtype.(others)...)
+    out = similar(copy(keys(d1), K), T)
     copyto!(out, d1)
-    merge!(out, d2)
+    merge!(out, others...)
     return out
 end
 
 if isdefined(Base, :mergewith) # Julia 1.5+
-    function Base.mergewith(combiner, d1::AbstractDictionary{K1, T1}, d2::AbstractDictionary{K2, T2}) where {K1, T1, K2, T2}
+    function Base.mergewith(combiner::Callable, d1::AbstractDictionary, others::AbstractDictionary...)
```

Comment:
I think it is best to get rid of the `Callable` constraint here - it doesn't seem to be recommended practice anymore as it restricts what kind of thing you might put here.

Reply:
E.g. in `Base`

```
ulia> methods(mergewith!)
# 4 methods for generic function "mergewith!" from Base:
 [1] mergewith!(combine)
     @ abstractdict.jl:287
 [2] mergewith!(combine, d1::Dict{K, V}, d2::AbstractDict) where {K, V}
     @ dict.jl:765
 [3] mergewith!(combine, d1::AbstractDict, d2::AbstractDict)
     @ abstractdict.jl:280
 [4] mergewith!(combine, d::AbstractDict, others::AbstractDict...)
     @ abstractdict.jl:276
```

Reply:
Done. Note that Callable is used in other places in the codebase, and should probably be removed at some point for consistency.